src/engine/Tableau.cpp

```diff
@@ -1704,6 +1704,47 @@ bool Tableau::allBoundsValid() const
     return _boundsValid;
 }
 
+// Ensure that non-basic variables are within bounds
+void Tableau::ensureNonBasicVariableGTLB( unsigned variable, double value )
+{
+    unsigned index = _variableToIndex[variable];
+    if ( !_basicVariables.exists( variable ) )
+        {
+            if ( FloatUtils::gt( value, _nonBasicAssignment[index] ) )
+                setNonBasicAssignment( variable, value, true );
+        }
+    else
+        {
+            // Recompute the status of an affected basic variable
+            // If the status changes, invalidate the cost function
+            unsigned oldStatus = _basicStatus[index];
+            computeBasicStatus( index );
+            if ( _basicStatus[index] != oldStatus )
+                _costFunctionManager->invalidateCostFunction();
+    }
+}
+
+
+// Ensure that non-basic variables are within bounds
+void Tableau::ensureNonBasicVariableLTUB( unsigned variable, double value )
+{
+    unsigned index = _variableToIndex[variable];
+    if ( !_basicVariables.exists( variable ) )
+        {
```

Comment:
Indentation is off, { should be under the "if"

Reply:
Fixed